experimental/rest/api.go

```diff
@@ -0,0 +1,113 @@
+package rest
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+// Client interface
+type Client interface {
+	Fetch(ctx context.Context, uriPath string, uriQuery string) (*http.Response, error)
+}
+
+// Input represents a rest api input
+type Input struct {
+	Name  string
+	Value string
+	Type  string
+}
+
+// API takes calls from the host, using the client to retrieve data from rest apis.
+type API struct {
+	Client         Client
+	Routes         map[string]string
+	DefaultParams  map[string]string
+	ErrorFormatter func(body io.ReadCloser) error
```

Comment:
`ErrorFormatter` shouldn't probably worry about closing the body, so I would change this to:

```suggestion
	ErrorFormatter func(body io.Reader) error
```